Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeSwitcher.cpp

```diff
@@ -423,4 +441,16 @@ namespace AzToolsFramework::ComponentModeFramework
                 }
             });
     }
+
+    void ComponentModeSwitcher::OnImGuiDropDownDisplayed()
+    {
+        ViewportUi::ViewportUiRequestBus::Event(
+            ViewportUi::DefaultViewportId, &ViewportUi::ViewportUiRequestBus::Events::HideSwitcher, m_switcherId);
+    }
+
+    void ComponentModeSwitcher::OnImGuiDropDownHidden()
+    {
+        ViewportUi::ViewportUiRequestBus::Event(
+            ViewportUi::DefaultViewportId, &ViewportUi::ViewportUiRequestBus::Events::ShowSwitcher, m_switcherId);
```

Comment:
question: This is probably fine now, but just something to be aware of, say in future we want to hide the Switcher for another reason, we might not want opening and closing ImGui to suddenly show the switcher, so just keep an eye on this is case as it can cause surprising, unwanted bugs in future 😛